src/modules/lesson/controllers/lesson.controller.spec.ts

```diff
@@ -152,7 +163,63 @@ describe('LessonController', () => {
 
       const returnValue = lessonController.readOneLesson(param, member);
 
-      expect(returnValue).toBeInstanceOf(ReadOneLessonResponseType);
+      expect(returnValue).toBeInstanceOf(ReadOneLessonDto);
+    });
+  });
+
+  describe('readSimilarLesson', () => {
+    let param: IdRequestParamDto;
+    let member: any;
+    let query: SimilarLessonQueryDto;
+    let mockSimilarLessons: SimilarLessonEntity;
+
+    beforeEach(async () => {
+      member = { id: faker.datatype.number() };
+      param = {
+        id: faker.datatype.number(),
+        model: 'lesson',
+      };
+      query = new SimilarLessonQueryDto();
+      mockSimilarLessons = plainToInstance(
+        SimilarLessonEntity,
+        JSON.parse(faker.datatype.json()),
+      );
+
+      mockLessonService.readSimilarLesson.mockReturnValue(mockSimilarLessons);
+    });
+
+    it('success - routing, plain object to class object converting', async () => {
+      const returnValue = await lessonController.readSimilarLesson(
+        param,
+        query,
+        member,
+      );
+
+      expect(mockLessonService.readSimilarLesson).toHaveBeenCalledTimes(1);
+      expect(mockLessonService.readSimilarLesson).toBeCalledWith(
+        param.id,
+        member.id,
+        query,
+      );
+      expect(returnValue).toBeInstanceOf(ReadSimilarLessonDto);
+      expect(returnValue.lessons).toStrictEqual(mockSimilarLessons);
+    });
+
+    it('success - plainToInstance Transform isBookmark', async () => {
+      mockSimilarLessons = plainToInstance(SimilarLessonEntity, {
+        isBookmark: faker.datatype.number(),
+      });
+      mockLessonService.readSimilarLesson.mockReturnValue(mockSimilarLessons);
+
+      const returnValue = await lessonController.readSimilarLesson(
+        param,
+        query,
+        member,
+      );
+
+      expect(
+        typeof returnValue.lessons['isBookmark'] === 'boolean',
+      ).toBeTruthy();
```

Comment:
```suggestion
      expect(typeof returnValue.lessons.isBookmark).toBe('boolean');
```
이렇게도 되겠네요

우선 [''] 이런식의 접근은 피해는게 좋을거같아요
우리가 순수 js 를 쓴다면 문제가 안되는데 typescript 를 쓴다면 어처리 타입으로 지정되면 dot 으로만 접근이 가능하고 dot 으로 접근이 안된다면 타입 지정을 잘못했나 확인해 보는게 좋을거같아요.

Reply:
괄호 표기법에서 점 표기법으로 수정하도록 하겠습니다!